src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -87,27 +86,31 @@ private void initBlock()
         setRegistryName(BLOCK_NAME);
         setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
         setCreativeTab(ModCreativeTabs.MINECOLONIES);
-        this.setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH).withProperty(VARIANT, EnumType.DEFAULT));
+        this.setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH).withProperty(VARIANT, RackType.DEFAULT));
         setHardness(BLOCK_HARDNESS);
         setResistance(RESISTANCE);
         setLightOpacity(LIGHT_OPACITY);
     }
 
-    @Override
-    public void getSubBlocks(final CreativeTabs itemIn, final NonNullList<ItemStack> items)
+    /**
+     * Check if a certain block should be replaced with a rack.
+     *
+     * @param block the block to check.
+     * @return true if so.
+     */
+    public static boolean shouldBlockBeReplacedWithRack(final Block block)
     {
-        items.add(new ItemStack(this, 1, EnumType.DEFAULT.getMetadata()));
+        return block == Blocks.CHEST || block == ModBlocks.blockRack;
     }
 
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
     @Override
-    public void onBlockPlacedBy(
-            final World worldIn, final BlockPos pos, final IBlockState state, final EntityLivingBase placer, final ItemStack stack)
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)